test/utils/TestConstantsAPI1812.scala

```diff
@@ -0,0 +1,351 @@
+    penaltyAmountAccruing = 123.45,
+    penaltyAmountPosted = 123.45,
+    penaltyChargeCreationDate = LocalDate.of(2022, 1, 1),
+    communicationsDate = LocalDate.of(2022, 1, 1),
+    penaltyChargeReference = "CHARGE123456",
+    penaltyChargeDueDate = LocalDate.of(2022, 2, 1),
+    appealStatus = Some("1"),
+    appealLevel = Some("1"),
+    principalChargeReference = "CHARGING12345",
+    principalChargeDueDate = LocalDate.of(2022, 3, 1)
+  )
+
+  val modelWithoutOptionalFields: LatePaymentPenalty = LatePaymentPenalty(
+    penaltyNumber = "1234ABCD",
+    penaltyCategory = LPPPenaltyCategoryEnum.firstPenalty,
+    penaltyStatus = LPPPenaltyStatusEnum.Posted,
+    penaltyAmountAccruing = 123.45,
+    penaltyAmountPosted = 123.45,
+    penaltyChargeCreationDate = LocalDate.of(2022, 1, 1),
+    communicationsDate = LocalDate.of(2022, 1, 1),
+    penaltyChargeReference = "CHARGE123456",
+    penaltyChargeDueDate = LocalDate.of(2022, 2, 1),
+    appealStatus = None,
+    appealLevel = None,
+    principalChargeReference = "CHARGING12345",
+    principalChargeDueDate = LocalDate.of(2022, 3, 1)
+  )
+
+  val lateSubmissionResponseModel: LateSubmission = LateSubmission(
+    lateSubmissionID = "ID123",
+    taxPeriod = Some("1"),
+    taxReturnStatus = "2",
+    taxPeriodStartDate = Some(LocalDate.of(2022, 1, 1)),
+    taxPeriodEndDate = Some(LocalDate.of(2022, 3, 31)),
+    taxPeriodDueDate = Some(LocalDate.of(2022, 5, 7)),
+    returnReceiptDate = Some(LocalDate.of(2022, 4, 1)),
+  )
+
+  val lateSubmissionModelNoOptional: LateSubmission = LateSubmission(
+    lateSubmissionID = "ID123",
+    None,
+    taxReturnStatus = "2",
+    None,
+    None,
+    None,
+    None
+  )
+
+  val lateSubmission: JsObject = Json.obj(
+    "lateSubmissionID" -> "ID123",
+    "taxPeriod" -> "1",
+    "taxReturnStatus" -> "2",
+    "taxPeriodStartDate" -> "2022-01-01",
+    "taxPeriodEndDate" -> "2022-03-31",
+    "taxPeriodDueDate" -> "2022-05-07",
+    "returnReceiptDate" -> "2022-04-01")
+
+  val lateSubmissionNoOption: JsObject = Json.obj(
+    "lateSubmissionID" -> "ID123",
+    "taxReturnStatus" -> "2")
+
+  val lspDetailsAllOptions: JsObject = Json.obj(
+    "penaltyNumber" ->"1234ABCD",
+    "penaltyOrder" -> "1",
+    "penaltyCategory" ->"P",
+    "penaltyStatus" ->"ACTIVE",
+    "penaltyCreationDate" ->"2022-01-01",
+    "penaltyExpiryDate" ->"2024-01-01",
+    "communicationsDate" -> "2022-01-01",
+    "lateSubmissions" -> Json.arr(lateSubmission),
+    "appealStatus" ->"1",
+    "appealLevel" ->"1",
+    "chargeReference" ->"foobar",
+    "chargeAmount" -> 123.45,
+    "chargeOutstandingAmount" -> 123.45,
+    "chargeDueDate" -> "2022-01-01"
+  )
+
+  val lspDetailsWithNoOptions: JsObject = Json.obj(
+    "penaltyNumber" ->"1234ABCD",
+    "penaltyOrder" -> "1",
+    "penaltyCategory" ->"P",
+    "penaltyStatus" ->"ACTIVE",
+    "penaltyCreationDate" ->"2022-01-01",
+    "penaltyExpiryDate" ->"2024-01-01",
+    "communicationsDate" -> "2022-01-01",
```

Comment:
Same comment as above regarding comma at the end of list. I think because this is inside a JSON object it wouldn't throw a scalastyle error but still best to sort it while we're here